components/admin_console/team_channel_settings/channel/details/channel_details.tsx

```diff
@@ -95,18 +104,25 @@ export default class ChannelDetails extends React.Component<ChannelDetailsProps,
 
         // If we don't have the team and channel on mount, we need to request the team after we load the channel
         if (!prevProps.team.id && !prevProps.channel.team_id && channel.team_id) {
-            this.props.actions.getTeam(channel.team_id);
+            this.props.actions.getTeam(channel.team_id).
+                then((data: any) => this.props.actions.loadScheme(data.data.scheme_id));
         }
     }
     async componentDidMount() {
-        const {channelID, channel, team, actions} = this.props;
-        actions.
+        const {channelID, channel, actions} = this.props;
+        await Promise.all([actions.
             getGroups(channelID).
             then(() => actions.getChannel(channelID)).
-            then(() => this.setState({groups: this.props.groups}));
-        if (!team.id && channel.team_id) {
-            actions.getTeam(channel.team_id);
-        }
+            then(() => this.setState({groups: this.props.groups})),
+        actions.getChannelModerations(channelID).
+            then(() => this.setState({channelPermissions: this.props.channelPermissions})),
+        actions.getTeam(channel.team_id).
+            then((data: any) => {
+                if (data.data) {
+                    this.props.actions.loadScheme(data.data.scheme_id);
+                }
+            }),
+        ]);
```

Comment:
Based on what i've seen done here the actions have been usually called in `componentDidMount` and then the selectors are called in the index and then pass the data down as props? 
Since these are all actions it seems okay to me?

Reply:
You're right, my mistake. Please disregard.